### DISCUSSION TITLE
Purpose of `tool.poetry.dependencies.python` if `package-mode = false`

### DISCUSSION THREAD

**Discussion by mo-lukecarr** [python-poetry/poetry#9261] [Q&A] [answered]
Is there any purpose to setting the Python version under `tool.poetry.dependencies` if I've set `package-mode` to `false`?

I'm trying to tidy up a repo, and I've noticed that the Python version is duplicated in both the `pyproject.toml` and `.python-version` files.

My (potentially incorrect) understanding is that the Python dependency is used when my package is consumed by others, ensuring that the consumer has a local Python installation compatible with my package.

For extra context: the repository in question is a private, deployable application, rather than a package.

With this in mind, is there any purpose to declaring the Python version in the dependencies section of my `pyproject.toml` file?

**Comment by dimbleby** [comment, marked as answer]
poetry still needs to know the range of supported python versions so that it can lock correctly

`.python-version` is nothing to do with poetry
